cli/cmd/timescaledb/connect.go

```diff
@@ -24,46 +23,44 @@ var timescaledbConnectCmd = &cobra.Command{
 }
 
 func init() {
-	timescaledbCmd.AddCommand(timescaledbConnectCmd)
+	Cmd.AddCommand(timescaledbConnectCmd)
 	timescaledbConnectCmd.Flags().BoolP("master", "m", false, "directly execute session on master node")
 	timescaledbConnectCmd.Flags().StringP("dbname", "d", "postgres", "database name to connect to")
+	timescaledbConnectCmd.Flags().StringP("user", "U", "", "database user to connect to")
 }
 
 func timescaledbConnect(cmd *cobra.Command, args []string) error {
-	var err error
-
-	var host, psqlCMD string
-
-	var master bool
-	master, err = cmd.Flags().GetBool("master")
+	master, err := cmd.Flags().GetBool("master")
 	if err != nil {
 		return fmt.Errorf("could not connect to TimescaleDB: %w", err)
 	}
 
 	dbname, err := cmd.Flags().GetString("dbname")
 	if err != nil {
-		return fmt.Errorf("could not change TimescaleDB password: %w", err)
+		return fmt.Errorf("could not connect to TimescaleDB: %w", err)
 	}
 
-	k8sClient := k8s.NewClient()
-	secret, err := k8sClient.KubeGetSecret(root.Namespace, root.HelmReleaseName+"-credentials")
+	dbuser, err := cmd.Flags().GetString("user")
 	if err != nil {
-		return fmt.Errorf("could not get TimescaleDB password: %w", err)
+		return fmt.Errorf("could not connect to TimescaleDB: %w", err)
 	}
 
-	dbDetails, err := common.FormDBDetails(user, dbname, root.Namespace, root.HelmReleaseName)
-	if err != nil {
-		return fmt.Errorf("could not get DB secret key from helm release: %w", err)
+	if dbuser == "" {
+		return fmt.Errorf("provide db-user to connect using -U flag or to use super-user use timescaledb superuser cmd")
 	}
 
-	user = dbDetails.User
-	var pass string
-	if bytepass, exists := secret.Data[dbDetails.SecretKey]; exists {
-		pass = string(bytepass)
-	} else {
-		return fmt.Errorf("could not get TimescaleDB password: %w", errors.New("user not found"))
+	dbDetails := &pgconn.DBDetails{
+		Namespace:   root.Namespace,
+		ReleaseName: root.HelmReleaseName,
+		DBName:      dbname,
+		User:        dbuser,
 	}
+	k8sClient := k8s.NewClient()
+	return PsqlConnect(k8sClient, dbDetails, master)
+}
 
+func PsqlConnect(k8sClient k8s.Client, dbDetails *pgconn.DBDetails, master bool) error {
+	var host, psqlCMD string
 	uri, err := utils.GetTimescaleDBURI(k8sClient, root.Namespace, root.HelmReleaseName)
 	if err != nil {
 		return err
```

Comment:
the uri isn't used in the "if master" branch so let's move this below that code to line 81